ui/ui-frontend/projects/archive-search/src/app/archive/archive-search/archive-search.component.ts

```diff
@@ -1068,6 +1074,14 @@ export class ArchiveSearchComponent implements OnInit, OnChanges, OnDestroy, Aft
     }
   }
 
+  public initArchiveUnitAllunitup(values: string[][]) {
+    let allUnitups: string[] = [];
+    values.forEach((value) => {
+      allUnitups = allUnitups.concat(value);
+    });
+    return [...new Set(allUnitups)];
```

Comment:
Ce ne serait pas équivalent à ça ?
```suggestion
    return [...new Set(values.flat())];
```